### PR TITLE
B-20440 Uploaded document viewing improvements - Add Rotate and Save Ability for Image Files

### PR DESCRIPTION
**Pull request by WeatherfordAaron**
implemented rotation and zoom capabilities for images

### CODE REVIEW

File: dist/28c58d40210ac28e2a3e.png
Comment (WeatherfordAaron) [inline dist/28c58d40210ac28e2a3e.png:1]: This file was auto-generated when I ran ```yarn build```

File: dist/index.js
Comment (WeatherfordAaron) [inline dist/index.js:1]: This file was auto-generated when I ran ```yarn build```

File: src/components/drivers/pdf-viewer.jsx
Comment (WeatherfordAaron) [inline src/components/drivers/pdf-viewer.jsx:1]: There were issues with the FileViewer trying to access something from an unmounted component, so this was added.

File: src/components/drivers/photo-viewer.jsx
Comment (WeatherfordAaron) [inline src/components/drivers/photo-viewer.jsx:1]: Not 100% sure why it removed the semi-colons from the ends of a lot of these lines. I believe that was an auto-format action.. however, I don't think it will interfere with the app's functionality.

File: webpack.config.dev.js
Comment (WeatherfordAaron) [inline webpack.config.dev.js]: This change was because the imported component started causing CircleCI to error out

File: dist/28c58d40210ac28e2a3e.png
Comment (taeJungCaci) [reply dist/28c58d40210ac28e2a3e.png:1]: I don't think we are supposed to include auto-generated items from doing `yarn build` do we?

File: webpack.config.dev.js
Comment (taeJungCaci) [inline webpack.config.dev.js:25]: Any idea why this change is needed now when it was working before? or was it never working?

Comment (taeJungCaci) [reply webpack.config.dev.js:25]: would the client error we were seeing in INT PRs about public path because of this change?

Comment (WeatherfordAaron) [reply webpack.config.dev.js:25]: it is not because of this change.. this change is because of those errors. However, I'm going to undo this one because it should work since it worked before

File: dist/28c58d40210ac28e2a3e.png
Comment (WeatherfordAaron) [reply dist/28c58d40210ac28e2a3e.png:1]: got rid of those changes

File: webpack.config.dev.js
Comment (WeatherfordAaron) [reply webpack.config.dev.js]: reverted that change

File: src/components/drivers/pdf-viewer.jsx
Comment (danieljordan-caci) [inline src/components/drivers/pdf-viewer.jsx:128]: Any reason you're not using strict equality here of `===`?

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Couple questions

File: src/components/drivers/pdf-viewer.jsx
Comment (WeatherfordAaron) [reply src/components/drivers/pdf-viewer.jsx]: old C# habits. Fix it.

File: src/components/drivers/photo-viewer.jsx
Comment (WeatherfordAaron) [reply src/components/drivers/photo-viewer.jsx]: You're right.. corrected

**Review by danieljordan-caci — COMMENTED** [review]
Is there any way to test this or is this just static code analysis for now?

**Comment by WeatherfordAaron** [comment]
> Is there any way to test this or is this just static code analysis for now?

Well for right now,  I think you can test it on integrationTesting.

**Comment by danieljordan-caci** [comment]
> I do see something else I probably need to change. In package.json, the version should step up, right? I believe the current way that the milmove app targets the v1.2.2 is from the tag that was added to the merge of "v1.2.2"... so I would probably need to add a tag of "v1.2.3" to this merge commit, I think as well
> 
> Alternatively, i could just remove the other v1.2.2 tag and make that tag be on this merge-commit

I believe this is correct, but I'll let @deandreJones or @cameroncaci chime in

I'll hold off approval until we hear back.

**Comment by WeatherfordAaron** [comment]
> > > I do see something else I probably need to change. In package.json, the version should step up, right? I believe the current way that the milmove app targets the v1.2.2 is from the tag that was added to the merge of "v1.2.2"... so I would probably need to add a tag of "v1.2.3" to this merge commit, I think as well
> > > Alternatively, i could just remove the other v1.2.2 tag and make that tag be on this merge-commit
> > 
> > 
> > I believe this is correct, but I'll let @deandreJones or @cameroncaci chime in
> > I'll hold off approval until we hear back.
> 
> if the tests prove successful, yes bump the version, then the corresponding mymove change should point to the new version

Sounds good. I bumped up the version in this package.json and after merge-commit goes into react-file-viewer main branch, I'll add that tag for the version.

**Review by danieljordan-caci — APPROVED** [review]
Able to rotate & save.

Kinda hate how you can't change the position of the view when zooming in (just kinda zooms in on center), but what can ya do. This meets the AC on the BL item.

https://github.com/user-attachments/assets/68c8e820-a0ec-4726-b832-cc81fb7e9b97

**Comment by WeatherfordAaron** [comment]
> Able to rotate & save.
> 
> Kinda hate how you can't change the position of the view when zooming in (just kinda zooms in on center), but what can ya do. This meets the AC on the BL item.
> 
>  Screen.Recording.2024-08-20.at.12.12.25.PM.mov

You can change position but you have to zoom in more.

@taeJungCaci found an issue where if you remove @transcom/react-file-viewer then ```git checkout package.json``` and then do a yarn install again, the controls disappear. If nobody has any suggestions as to why that happens, I'll be investigating now.

**Review by taeJungCaci — APPROVED** [review]

**Pull request merged**
